packages/markdown/remark/src/rehype-images.ts

```diff
@@ -3,17 +3,28 @@ import type { MarkdownVFile } from './types.js';
 
 export function rehypeImages() {
 	return () =>
-		function (tree: any, file: MarkdownVFile) {
-			visit(tree, (node) => {
-				if (node.type !== 'element') return;
-				if (node.tagName !== 'img') return;
+			function (tree: any, file: MarkdownVFile) {
+					const imageOccurrenceMap = new Map();
 
-				if (node.properties?.src) {
-					if (file.data.imagePaths?.has(node.properties.src)) {
-						node.properties['__ASTRO_IMAGE_'] = node.properties.src;
-						delete node.properties.src;
-					}
-				}
-			});
-		};
+					visit(tree, (node) => {
+							if (node.type !== 'element') return;
+							if (node.tagName !== 'img') return;
+
+							if (node.properties?.src) {
+									if (file.data.imagePaths?.has(node.properties.src)) {
+											const { alt, ...otherProps } = node.properties;
+
+											// Initialize or increment occurrence count for this image
+											const index = imageOccurrenceMap.get(node.properties.src) || 0;
+											imageOccurrenceMap.set(node.properties.src, index + 1);
+
+											node.properties['__ASTRO_IMAGE_'] = JSON.stringify({ ...otherProps, index });
+											
+											Object.keys(otherProps).forEach((prop) => {
+													delete node.properties[prop];
+											});
```

Comment:
This is a little bit concerning, but normally it's fine because image services **should** re-output everything they received (but didn't understand) as attributes. It's non-breaking for the built-in image services at least.

In the future, we should most likely offer an option to resolve the img's `src`, but otherwise leave the `img` tag alone, just so that people who want/need full control can do so.

Reply:
Yeah you're right I thought the same what if it doesn't return all the random extra stuff so far it has but yeah adding in ability to opt out or use a service that wouldn't return extra stuff 

I guess this would happen in the remark-collect-images shouldOptimize function 

But you're right it's pretty ugly and weird to stuff everything inside __ASTRO_IMAGE__ and assume it comes out of getImage()

Reply:
Yeah, ideally we'd do it in Node, but last time I tried it didn't work. Maybe something to revisit in the future. It works at the moment, and that's what's important so far, so it's ok